.buildkite/pipeline.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+##
+# Dynamic Buildkite pipeline generator.
+##
+#
+# It outputs valid Buildkite pipeline in YAML format.
+#
+# To use it, define the following Steps under your Buildkite's Pipeline Settings:
+#
+# steps:
+#   - command: .buildkite/pipeline.sh | buildkite-agent pipeline upload
+#     label: ":pipeline: Upload"
+#
+# For more details, see:
+# https://buildkite.com/docs/pipelines/defining-steps#dynamic-pipelines.
+#
+
+set -eux
+
+# Helper that ensures the build is triggered for a pull request and that there
+# are no code-related changes compared to the pull request's base branch.
+pr_and_no_code_related_changes() {
+  # Check if the build was triggered for a pull request.
+  if [[ -z $BUILDKITE_PULL_REQUEST_BASE_BRANCH ]]; then
```

Comment:
can't skip in this case:
- we should build on master for PR merge commits, because there are steps that are set to run on master only
- you'd normally push a branch first and then open a PR for it. that first push wouldn't be built under this rule, because a PR doesn't exist yet